src/mono/mono.proj

```diff
@@ -1101,8 +1066,6 @@
           DestinationFiles="@(_MonoRuntimeBuildArtifacts->'$(RuntimeBinDir)build\%(RecursiveDir)%(Filename)%(Extension)')"
           SkipUnchangedFiles="true"
           Condition="'$(BuildMonoAOTCrossCompilerOnly)' != 'true'" />
-
-    <Exec Condition="'$(BuildMonoAOTCrossCompilerOnly)' != 'true' and '$(MonoGenerateOffsetsOSGroups)' == '' and ('$(TargetsOSX)' == 'true' or '$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true')" Command="install_name_tool -id @rpath/$(MonoFileName) $(RuntimeBinDir)$(MonoFileName)" />
```

Comment:
did you verify that the rpath is correct after this change?

Reply:
I confirmed this works because we pass `-DMONO_SHARED_LIB_NAME=$(MonoSharedLibName)` into cmake now so the filename is correct from the beginning (and thus the rpath).

Reply:
TBH, I mainly relied upon the CI for this one. We switched cmake `install` with our wrapper `install_with_stripped_symbols`, which is used for all binaries for coreclr, libs and corehost.

Now that you have mentioned it, I tested it on osx-arm64 and `otool` gives me the same output before (main) and after (PR):

```sh
$ ./build.sh mono -c Release

$ otool -l artifacts/bin/mono/osx.arm64.Release/libcoreclr.dylib
[...]
Load command 4
          cmd LC_ID_DYLIB
      cmdsize 48
         name @rpath/libcoreclr.dylib (offset 24)
   time stamp 1 Thu Jan  1 02:00:01 1970
      current version 2.0.0
[...]
```
If the passing CI legs do not indicate that iOS / other targets are in the clear and this testing (on osx-arm64) is not enough, I can try to build for those platforms as well?

Reply:
I looked through the failures on runtime-extra-platforms and none of them seem to be related :)